src/rest/connector/libs/nd/implementation.py

```diff
@@ -0,0 +1,310 @@
+    Implementation of Rest connection to devices based on pyATS BaseConnection
+    for ND
+
+    YAML Example
+    ------------
+
+        devices:
+            nd1:
+                platform: nd
+                connections:
+                    rest:
+                        class: rest.connector.Rest
+                        ip : "2.3.4.5"
+                credentials:
+                    rest:
+                        username: admin
+                        password: cisco123
+
+    Code Example
+    ------------
+        >>> from pyats.topology import loader
+        >>> testbed = loader.load('/users/xxx/xxx/testbed.yaml')
+        >>> device = testbed.devices['nd1']
+        >>> device.connect(alias='rest', via='rest')
+        >>> device.rest.connected
+        True
+    """
+
+    @BaseConnection.locked
+    def connect(self, timeout=30, retries=3, retry_wait=10):
+        """connect to the device via REST
+        Arguments
+        ---------
+            timeout (int): Timeout value
+            retries (int): Max retries on request exception (default: 3)
+            retry_wait (int): Seconds to wait before retry (default: 10)
+
+        Raises
+        ------
+
+        Exception
+        ---------
+            If the connection did not go well
+
+        Note
+        ----
+        There is no return from this method. If something goes wrong, an
+        exception will be raised.
+
+        YAML Example
+        ------------
+            devices:
+                nd1:
+                    platform: nd
+                    connections:
+                        rest:
+                            class: rest.connector.Rest
+                            ip : "2.3.4.5"
+                    credentials:
+                        rest:
+                            username: admin
+                            password: cisco123
+
+        Code Example
+        ------------
+            >>> from pyats.topology import loader
+            >>> testbed = loader.load('/users/xxx/xxx/testbed.yaml')
+            >>> device = testbed.devices['nd1']
+            >>> device.connect(alias='rest', via='rest')
+        """
+
+        if self.connected:
+            return
+
+        if 'host' in self.connection_info:  # the dictionary of connection info based on path (via)
+            ip = self.connection_info['host']
+        else:
+            ip = self.connection_info['ip'].exploded
+        if 'port' in self.connection_info:
+            port = self.connection_info['port']
+            self.url = 'https://{ip}:{port}/'.format(ip=ip, port=port)
+        else:
+            self.url = 'https://{ip}/'.format(ip=ip)
+        login_url = '{f}login'.format(f=self.url)  # CHANGE: login url
+
+        username, password = get_username_password(self)  # gets credentials from rest
+
+        payload = {
+            "userName": username,
+            "userPasswd": password,
+            "domain": "DefaultAuth"  # CHANGE: domain
```

Comment:
what `# CHANGE: domain` mean?

Reply:
Sorry, some comments from the initial documentation / implementation phase slipped through.
Removed them.